hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/workflow/diagnostic/DiagnosticFileManager.java

```diff
@@ -0,0 +1,129 @@
+    }
+
+    public File getExtractionDirectory() {
+        return extractionDirectory;
+    }
+
+    public File getLogDirectory() {
+        return logDirectory;
+    }
+
+    public void init(final File outputDirectory, final File bdioDirectory, final String runId) {
+        reportDirectory = new File(new File(outputDirectory, "reports"), runId);
+        relevantDirectory = new File(new File(outputDirectory, "relevant"), runId);
+        extractionDirectory = new File(new File(outputDirectory, "extractions"), runId);
+        logDirectory = new File(new File(outputDirectory, "logs"), runId);
+
+        trackedDirectories.add(bdioDirectory);
+        trackedDirectories.add(reportDirectory);
+        trackedDirectories.add(relevantDirectory);
+        trackedDirectories.add(extractionDirectory);
+        trackedDirectories.add(logDirectory);
+
+        for (final File file : trackedDirectories) {
+            logger.info("Creating diagnostics directory: " + file.getPath());
+            file.mkdirs();
+        }
+    }
+
+    public void cleanup() {
+        try {
+            for (final File file : trackedDirectories) {
+                logger.info("Cleaning diagnostics directory: " + file.getPath());
+                FileUtils.deleteDirectory(file);
+            }
+        } catch (final IOException e) {
+            logger.error("Failed to cleanup:");
+            e.printStackTrace();
+        }
+    }
+
+    public List<File> getAllDirectories() {
+        return trackedDirectories;
+    }
+
+    public void registerFileOfInterest(final ExtractionId extractionId, final File file) {
+        registerFileOfInterest(file, extractionId.toUniqueString());
+    }
+
+    public void registerGlobalFileOfInterest(final File file) {
+        registerFileOfInterest(file, "global");
+    }
+
+    private void registerFileOfInterest(final File file, final String directoryName) {
+        try {
+            if (file == null) {
+                return;
+            }
+            if (isChildOfTrackedFolder(file)) {
+                logger.info("Asked to track file '" + file.getPath() + "' but it is already being tracked.");
+                return;
+            }
+            if (file.isFile()) {
+                final File dest = findNextAvailableRelevant(directoryName, file.getName());
+                FileUtils.copyFile(file, dest);
+            } else if (file.isDirectory()) {
+                final File dest = findNextAvailableRelevant(directoryName, file.getName());
+                FileUtils.copyDirectory(file, dest);
+            }
+        } catch (final Exception e) {
+            logger.trace("Failed to copy file to relevant directory:" + file.toString());
+        }
+    }
+
+    private boolean isChildOfTrackedFolder(final File file) {
+        for (final File trackedFile : trackedDirectories) {
+            if (file.toPath().startsWith(trackedFile.toPath())) {
+                return true;
+            }
+        }
```

Comment:
Would streams be a cleaner way to do this?
return trackedDirectories.stream().anyMatch(file.toPath().startsWith(trackedFile.toPath));

Reply:
Done